### PR TITLE
fix: Fix `Foldable1 (Coyoneda f)` instance

### PR DESCRIPTION
**Pull request by srghma**
requires https://github.com/purescript/purescript-lazy/pull/30

### CODE REVIEW

**Review by JordanMartinez — APPROVED** [review]
Not sure whether it would be more efficient to handwrite the implementation for `foldr1`/`foldl1`. I'll let other maintainers comment.

**Comment by srghma** [comment]
I actually tried to rewrite it with `foldXXX1` of `f`, but didnt manage to solve

**Comment by JordanMartinez** [comment]
Included in #106

**Pull request closed without merging**
